packages/e2e/test/artillery/StakePoolSearch.ts

```diff
@@ -0,0 +1,99 @@
+import { AfterResponse, BeforeRequest, WhileTrue } from './artillery';
+import { Cardano, Paginated, QueryStakePoolsArgs } from '@cardano-sdk/core';
+import { logger } from '@cardano-sdk/util-dev';
+
+interface StakePoolSearchVars extends Paginated<Cardano.StakePool> {
+  args: [QueryStakePoolsArgs];
+  moreResults: boolean;
+  queryFirst: boolean;
+  queryWithName: boolean;
+  queryWithStatus: boolean;
+  startedAt: number;
+}
+
+const PAGE_SIZE = 100;
+
+const names: Record<string, boolean> = {};
+let firstUser = '';
+
+const getStatName = (vars: StakePoolSearchVars) => {
+  if (vars.queryFirst) return 'query.full_first';
+  if (vars.queryWithStatus) return vars.queryWithName ? 'query.name_status' : 'query.status';
+  return vars.queryWithName ? 'query.name' : 'query.full';
+};
+
+export const after: AfterResponse<StakePoolSearchVars> = async (_params, _response, ctx, ee, done) => {
+  // Take values to randomize next requests
+  try {
+    for (const result of ctx.vars.pageResults) if (result.metadata) names[result.metadata.name] = true;
+  } catch (error) {
+    logger.fatal(error, ctx, ctx.vars);
+  }
+
+  // Send custom metrics
+  ee.emit('histogram', `${getStatName(ctx.vars)}.time`, Date.now() - ctx.vars.startedAt);
+
+  done();
+};
+
+export const before: BeforeRequest<StakePoolSearchVars> = async (_params, ctx, ee, done) => {
+  let options = ctx.vars.args?.[0];
```

Comment:
The arguments structure of our providers is changed from `args:[arg1, arg2]` to `{ arg1, arg2}`
PR for a reference: https://github.com/input-output-hk/cardano-js-sdk/pull/393 @iccicci

Reply:
10e1a9b97 should fix this